src/Enum.php

```diff
@@ -205,8 +224,8 @@ protected static function resolve(): array
         }
 
         foreach (self::$cache[$class] as $name => $enum) {
-            self::$cache[$class][$name]['value'] = static::make($name)->getValue();
-            self::$cache[$class][$name]['index'] = static::make($name)->getIndex();
+            self::$cache[$class][$name]['value'] = static::getMappedValue($name) ?? static::make($name)->getValue();
```

Comment:
Now that we have three keys, I wonder whether an object could add some clarity here. It could be as simple as 

```php
class EnumDefinition
{
    public $name;
    public $index;
    public $value;
}
```

What do you think?

Reply:
Yes, we can do so - a simple property holder class. Will change it all over the enum class.

Reply:
Okay, got to some problem: the `EnumDefinition` couldn't be cached in a simple format (array, json). I also don't see a big benefit of using it(?). So is it worth the trade?